pkg/v1/tkg/test/tkgctl/shared/package.go

```diff
@@ -0,0 +1,137 @@
+)
+
+const (
+	getResourceTimeout  = time.Minute * 1
+	waitForReadyTimeout = time.Minute * 10
+	pollingInterval     = time.Second * 30
+)
+
+// create cluster client from kubeconfig
+func createClientFromKubeconfig(exportFile string, scheme *runtime.Scheme) (client.Client, error) {
+	config, err := clientcmd.LoadFromFile(exportFile)
+	Expect(err).ToNot(HaveOccurred(), "Failed to load cluster Kubeconfig file from %q", exportFile)
+
+	rawConfig, err := clientcmd.Write(*config)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create raw config ")
+
+	restConfig, err := clientcmd.RESTConfigFromKubeConfig(rawConfig)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create rest config ")
+
+	client, err := client.New(restConfig, client.Options{Scheme: scheme})
+	Expect(err).ToNot(HaveOccurred(), "Failed to create a cluster client")
+
+	return client, nil
+}
+
+func checkClusterCBS(ctx context.Context, mccl, wccl client.Client, mcClusterName, wcClusterName, infrastructureName string) error {
+	log.Infof("Verify addons on workload cluster %s with management cluster %s", wcClusterName, mcClusterName)
+
+	// get ClusterBootstrap and return error if not found
+	clusterBootstrap := getClusterBootstrap(ctx, mccl, constants.TkgNamespace, mcClusterName)
+
+	// verify cni package is installed on the workload cluster
+	cniPkgShortName, cniPkgName, cniPkgVersion := getPackageDetailsFromCBS(clusterBootstrap.Spec.CNI.RefName)
+	verifyPackageInstall(ctx, wccl, addonutil.GeneratePackageInstallName(wcClusterName, cniPkgShortName), cniPkgName, cniPkgVersion)
+
+	// verify the remote kapp-controller package is installed on the management cluster
+	kappPkgShortName, kappPkgName, kappPkgVersion := getPackageDetailsFromCBS(clusterBootstrap.Spec.Kapp.RefName)
+	verifyPackageInstall(ctx, mccl, addonutil.GeneratePackageInstallName(wcClusterName, kappPkgShortName), kappPkgName, kappPkgVersion)
+
+	// verify csi and cpi package is installed on the workload cluster if in vSphere environment
+	if infrastructureName == "vsphere" {
+		csiPkgShortName, csiPkgName, csiPkgVersion := getPackageDetailsFromCBS(clusterBootstrap.Spec.CSI.RefName)
+		verifyPackageInstall(ctx, wccl, addonutil.GeneratePackageInstallName(wcClusterName, csiPkgShortName), csiPkgName, csiPkgVersion)
+
+		cpiPkgShortName, cpiPkgName, cpiPkgVersion := getPackageDetailsFromCBS(clusterBootstrap.Spec.CPI.RefName)
+		verifyPackageInstall(ctx, wccl, addonutil.GeneratePackageInstallName(wcClusterName, cpiPkgShortName), cpiPkgName, cpiPkgVersion)
+	}
+
+	// loop over additional packages list in clusterBootstrap spec to check package information
+	if clusterBootstrap.Spec.AdditionalPackages != nil {
+		// validate additional packages
+		for _, additionalPkg := range clusterBootstrap.Spec.AdditionalPackages {
+			pkgShortName, pkgName, pkgVersion := getPackageDetailsFromCBS(additionalPkg.RefName)
+
+			// TODO: temporarily skip verifying tkg-storageclass due to install failure issue.
```

Comment:
Address in follow up

Reply:
Do we need to find a ticket tracking this?